src/Logary/HealthCheck.fs

```diff
@@ -42,96 +38,89 @@ module HealthCheck =
   open System
   open System.Runtime.CompilerServices
 
-  open FSharp.Actor
+  open Hopac
 
   open Logary.Internals
-  open Logary.Measure
 
   /// A key in the `data` map inside the `Measure` type.
   [<Literal>]
   let Description = "description"
 
-  /// A key in the `data` map inside the `Measure` type.
-  [<Literal>]
-  let Exception = "exception"
-
   /// Sets the description property of the measurement's data map
   [<CompiledName "SetDescription">]
-  let setDesc description m =
-    m |> setData Description description
-
-  /// Sets the exception property of the measurement's data map
-  [<CompiledName "SetException">]
-  let setExn e m =
-    m |> setData Exception e
+  let setDesc (description: string) m =
+    Message.field Description description m
 
   /// Tries to get the description value from the measure.
   [<CompiledName "TryGetDescription">]
   let tryGetDesc m =
-    m.m_data |> Map.tryFind Description |> Option.fold (fun s t -> t :?> string) ""
-
-  /// Tries to get an exception from the measure
-  [<CompiledName "TryGetException">]
-  let tryGetExn m =
-    m.m_data |> Map.tryFind Exception |> Option.map (fun x -> x :?> exn)
+    match Message.tryGetField Description m with
+    | (Some (Field (String s, _))) ->  s
+    | _ -> ""
 
   /// An implementation of the ResultData interface that wraps a Measure and
   /// uses its 'data' Map to read.
-  type MeasureWrapper(m : _) =
+  type MeasureWrapper(m : Message) =
     interface ResultData with
       member x.Measure     = m
       member x.Description = tryGetDesc m
-      member x.Exception   = tryGetExn m
+      //member x.Exception   = tryGetExn m
     override x.ToString() =
-      sprintf "HealthCheck(name=%s, exn=%A, value=%A, level=%A)"
-        m.m_path.joined (tryGetExn m) m.m_value m.m_level
+      sprintf "HealthCheck(name=%s, value=%A, level=%A)"
+        (PointName.joined m.name) m.value m.level
 
-  module Measure =
+  module Message =
     /// Transform the measure to a HealthCheck.ResultData.
     let toResult (m : _) =
       MeasureWrapper m :> ResultData
       |> HasValue
 
   type HealthCheckMessage =
-    | GetResult of HealthCheckResult Types.ReplyChannel
-    | ShutdownHealthCheck of Acks Types.ReplyChannel
+    | GetResult of Ch<HealthCheckResult>
+    | ShutdownHealthCheck of IVar<Acks>
 
   type HealthCheckInstance =
-    { actor : HealthCheckMessage IActor }
+    { reqCh      : HealthCheckMessage Ch }
 
   type private FnCheckerState =
     { last : HealthCheckResult }
 
-  let private mkFromFunction fn =
-    (fun (inbox : IActor<_>) ->
-      let rec running state = async {
-        let! msg, mopts = inbox.Receive()
-        match msg with
-        | GetResult chan ->
+  let private mkFromFunction (fn : unit -> Job<HealthCheckResult>) =
+    let ch = { reqCh = Ch.Now.create () }
+
+    let rec running _ = job {
+        let! req = Ch.take ch.reqCh
```

Comment:
This is fine for requests, because they won't ever overwhelm the system. For input, however, we should use bounded message buffers.